classes/core/PKPEncryptionServiceProvider.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/**
+ * @file classes/core/PKPEncryptionServiceProvider.php
+ *
+ * Copyright (c) 2024 Simon Fraser University
+ * Copyright (c) 2024 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class PKPEncryptionServiceProvider
+ *
+ * @brief Register encryption service and validate app key.
+ */
+
+namespace PKP\core;
+
+use APP\core\Application;
+use Illuminate\Encryption\EncryptionServiceProvider as IlluminateEncryptionServiceProvider;
+
+class PKPEncryptionServiceProvider extends IlluminateEncryptionServiceProvider
+{
+    /**
+     * @copydoc Illuminate\Encryption\EncryptionServiceProvider::registerEncrypter()
+     */
+    protected function registerEncrypter()
+    {
+        // if application not installed, we do not check for app key as
+        // may be it's need to be generated at application installation process
+        if (!Application::isInstalled()) {
+            return;
+        }
+
+        // if no app key set, the encrypter can not be registered
+        if (!PKPAppKey::hasKey()) {
+            return;
+        }
```

Comment:
This is to allow the application to boot and run if `app_key` not set and we don't want to make it a must requirement for app to run . We can remove it if we decide to it make a must .

Reply:
Because we're adding the app key automatically both on install and upgrade, I see no downsides to making the app key a requirement right now. It'll become necessary later as we adopt more of Laravel, so we may as well add the requirement now, since it's low burden to the admin. It'll keep our code simpler.